Sources/DVR/Interaction.swift

```diff
@@ -106,5 +109,8 @@ extension Interaction {
         self.response = HTTPURLResponse(dictionary: response)
         self.recordedAt = Date(timeIntervalSince1970: recordedAt)
         self.responseData = Interaction.dencodeBody(response["body"], headers: response["headers"] as? [String: String])
+        self.filter = filter
+        self.filter?.beforeRecordRequest(&self.request)
+        self.filter?.beforeRecordResponse(&self.response,&self.responseData)
```

Comment:
From my understanding of the framework, `beforeRecordRequest` should be called just before looking for an interaction match, while `beforeRecordResponse` should be called just before persisting a response. 

If an array of Interactions is initialized from a saved recording, everything should already be scrubbed, so you shouldn't need to run these methods at all in the initializer

Reply:
This is definitely something I want to talk about on our call.